File: Water/H5Api/src/main/OpenDataApi/models/RegExp.java
package OpenDataApi.models;

import OpenDataAccess.Entity.*;
import OpenDataAccess.LambdaInterface.IFunction2;

import java.util.Date;
import java.util.List;

@ITablePropertyAttribute(Name = "t_d_RegExp2", PrimaryKey = "Id")
public class RegExp extends EntityModel implements IEntity {
    public String Id = null;
    public String Name = null;
    public Date Create_Date = null;
    public String Expression = null;
    public String Remark = null;

    @Override
    public void InsertValidate(List<IFunction2<IValidate, IEntityData, String>> validateList) {
        validateList.add(this.InsertValidateUnique(RegExp.class, "Name", "对不起，该名称已存在！"));
    }

    @Override
    public void UpdateValidate(List<IFunction2<IValidate, IEntityData, String>> validateList) {
        validateList.add(this.UpdateValidateUnique(RegExp.class, "Name", "对不起，该名称已存在！"));
    }
}

File: Water/Web/configs/entities/RegExp.js
const EntityModelsConfig = require("./EntityModelsConfig");

module.exports = {
    Name: "RegExp",
    PrimaryKey: "Id",
    Properties: GetProperties(),
    ModelsConfig: EntityModelsConfig("RegExp")
}

function GetProperties() {
    return [
        GetProperty("Id", "Id"),
        GetProperty("Name", "名称"),
        GetProperty("Expression", "表达式"),
        GetProperty("Remark", "备注"),
        GetProperty("Create_Date", "创建时间")
    ]
}

function GetProperty(Name, Label) {
    return { Name, Label }
}